src/wallet/scriptpubkeyman.cpp

```diff
@@ -298,7 +298,22 @@ std::unique_ptr<SigningProvider> LegacyScriptPubKeyMan::GetSigningProvider(const
 
 bool LegacyScriptPubKeyMan::CanProvide(const CScript& script, SignatureData& sigdata)
 {
```

Comment:
Of course there is no pre-box equivalent, CWallet was just passed in as the SiginingProvider always!

Reply:
This function has no tests, half a dozen different code paths and calls really confusing legacy code like IsMine. There's probably less than 5 people on the planet who understand it, and even fewer who understand the ways this can break for some weird use cases. The only excuse for not documenting it is if it's copied from a legacy function.  
<img width="775" alt="Schermafbeelding 2019-09-18 om 09 18 31" src="https://user-images.githubusercontent.com/10217/65124848-4fc47200-d9f5-11e9-8772-b278589d4f46.png">

Reply:
I've added some comments.